### ISSUE TITLE
Review performance

### ISSUE THREAD

**Issue by roberttod** [QubitProducts/driftwood#18]
With logging in jolt + queue-that, driftwood roughly doubles execution time when emitting a large number of events **when logging is disabled**. I think the solution would probably be not to loop over all the levels every time a log happens, which seems to be a major factor.

@oliverwoodings I can show you how I was able to record the performance change if you would like.

**Comment by oliverwoodings** [comment]
Good lord, that isn't good. I'll look into it first thing tomorrow! Yeah
i'd love to see how you measured it.

**Comment by oliverwoodings** [comment]
I've set up some benchmarking (PR inbound). The variants here involve adding a return statement into [this function](https://github.com/QubitProducts/driftwood/blob/master/src/createLogger.js#L10) at various points. Here is a subset of the output:

#### control:
```
logging when disabled: basic at 8984443 ops/sec
logging when disabled: fancy at 9166139 ops/sec
```

#### no indexOf and no each:
```
logging when disabled: basic at 25742866 ops/sec
logging when disabled: fancy at 21824576 ops/sec
```

#### indexOf, no each:
```
logging when disabled: basic at 8275225 ops/sec
logging when disabled: fancy at 7545206 ops/sec
```

I'll come up with a way to check the levels without using `indexOf`

**Comment by oliverwoodings** [comment]
Made a few tweaks, managed to get disabled perf up even more:

```
Chrome 50.0.2661 (Mac OS X 10.11.1)  logging when disabled: basic at 32787118 ops/sec
Chrome 50.0.2661 (Mac OS X 10.11.1)  logging when disabled: fancy at 31555893 ops/sec
```

**Comment by roberttod** [comment]
Great work @oliverwoodings! I will test the performance in Jolt again to see if it improved.

**Comment by oliverwoodings** [comment]
It's not actually released gimme a bit :p

**Comment by roberttod** [comment]
I just realised, was about to ask where the commit was...

**Comment by oliverwoodings** [comment]
Made it a bit faster again:

```
Chrome 50.0.2661 (Mac OS X 10.11.1)  logging when disabled: basic at 1252929083 ops/sec
Chrome 50.0.2661 (Mac OS X 10.11.1)  logging when disabled: fancy at 108260588 ops/sec
```

38x faster, lol

**Comment by oliverwoodings** [comment]
Closed by #20
